src/Microsoft.TemplateEngine.Edge/Constraints/OSConstraint.cs

```diff
@@ -43,22 +51,82 @@ internal OSConstraint(IEngineEnvironmentSettings environmentSettings, ITemplateC
             {
-                IEnumerable<OSPlatform> supportedOS = ParseArgs(args);
-
-                foreach (OSPlatform platform in supportedOS)
+                try
                 {
-                    if (RuntimeInformation.IsOSPlatform(platform))
+                    IEnumerable<OSPlatform> supportedOS = ParseArgs(args);
+
+                    foreach (OSPlatform platform in supportedOS)
                     {
-                        return new TemplateConstraintResult(TemplateConstraintResult.Status.Allowed);
+                        if (RuntimeInformation.IsOSPlatform(platform))
+                        {
+                            return new TemplateConstraintResult(TemplateConstraintResult.Status.Allowed);
+                        }
                     }
+                    //TODO: localize
+                    return new TemplateConstraintResult(TemplateConstraintResult.Status.Restricted, $"Running template on {RuntimeInformation.OSDescription} is not supported, supported OS is/are: {string.Join(", ", supportedOS)}.");
+                }
+                catch (ConfigurationException ce)
+                {
+                    return new TemplateConstraintResult(TemplateConstraintResult.Status.NotEvaluated, ce.Message, "Check the constraint configuration in template.json");
                 }
-                //TODO: localize
-                return new TemplateConstraintResult(TemplateConstraintResult.Status.Restricted, $"Running template on {RuntimeInformation.OSDescription} is not supported, supported OS is/are: {string.Join(", ", supportedOS)}.");
             }
 
+            //supported configuration:
+            // "args": "Windows"
+            // "args": [ "Linux", "Windows" ]
             private static IEnumerable<OSPlatform> ParseArgs(string? args)
             {
-                throw new NotImplementedException();
+                string supportedValues = string.Join(", ", _platformMap.Keys.Select(e => $"'{e}'"));
+                if (string.IsNullOrWhiteSpace(args))
+                {
+                    throw new ConfigurationException("Argument(s) were not specified. At least one argument should be specified.");
+                }
+
+                JToken? token;
+                try
+                {
+                    token = JToken.Parse(args!);
+                }
+                catch (Exception e)
+                {
+                    throw new ConfigurationException($"'{args}' is not a valid JSON.", e);
+                }
+
+                if (token!.Type == JTokenType.String)
+                {
+                    return new[] { Parse(token.Value<string>()) };
+                }
+                else if (token is JArray jArray)
+                {
+                    IEnumerable<string?> values = jArray.Values<string>();
+                    List<OSPlatform> readValues = new List<OSPlatform>();
+                    foreach (string? value in values)
```

Comment:
Does the ConstraintManager validate the constraint arg schema validity? I'm just wondering what happens if array of non-strings would lead to meaningful error message.

In any case - it is probably worth to error out on empty array

Reply:
Added the check for empty array. 
There is no validation against schema at all unfortunately, it's part of https://github.com/dotnet/templating/issues/2594.